Horizon/Horizon/Sources/Features/Notebook/Common/Data/CourseNotesRepository.swift

```diff
@@ -0,0 +1,214 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import Combine
+
+struct CourseNote {
+    let id: String
+    let date: Date
+    let content: String
+    let institution: String
+    let courseId: String
+    let course: String
+    let labels: [String] // e.g. "Important", "Confusing", etc.
+}
+
+struct CourseNotesRepository {
+    func get() -> AnyPublisher<[CourseNote], any Error> {
+        return Just(notes).setFailureType(to: Error.self).eraseToAnyPublisher()
+    }
+
```

Comment:
Here, you want to set mock data. However, you don't need to create a CourseNotesRepository for it. Feel free to delete the CourseNotesRepository and add your mock data directly inside the CourseNote model.
```
var mock: [CourseNote] = [
CourseNote(id: UUID().uuidString,
                   date: Date(timeIntervalSinceNow: -10000),
                   content: "This is a note 1",
                   institution: "Brigham Young University",
                   courseId: "1", course: "CS193P",
                   labels: ["Important"])
]
```

Reply:
I would prefer not to do that. The repository is just a source for data. In the future it'll change from hardcoded data to some other source. I'd prefer not to clutter the data object with test data.

Reply:
I agree with you for the functionality for repo, but if you check our structure we did't use any repo, just use interactor for getting data from local or remote using `ReactiveStore` and you can check the implementation here `GetProgramsInteractorLive`